packages/e2e-tests/pages/patients/LabRequestPage/modals/LabRequestModalBase.ts

```diff
@@ -76,4 +160,101 @@ export class LabRequestModalBase {
   getCurrentDateTime(): string {
     return format(new Date(), "yyyy-MM-dd'T'HH:mm");
   }
+
+  // Generic method to select items by text and return their categories
+  async selectItemsByText(itemNames: string[]) {
+    const itemCategories: string[] = [];
+    for (const itemName of itemNames) {
+      const item = this.listItems.filter({ hasText: itemName }).nth(0);
+      const itemCategory = await item.locator('..').getByTestId(CATEGORY_TEXT_TEST_ID).nth(0).textContent();
+      itemCategories.push(itemCategory || '');
+      await item.click();
+    }
+    return itemCategories; 
+  }
+
+  // Generic method to validate selected items and their categories in a table
+  async validateSelectedItemsAndCategoriesInTable(
+    expectedItems: string[], 
+    expectedCategories: string[]
+  ) {
+    // Wait for the selected items table to be visible
+    await this.selectedItemsList.first().waitFor({ state: 'visible' });
+    
+    // Get all selected items in the table
+    const actualCount = await this.selectedItems.count();
+    
+    // Verify the count matches
+    await expect(actualCount).toBe(expectedItems.length);
+    
+    // Verify each expected item is present in the table
+    for (let i = 0; i < expectedItems.length; i++) {
+      const expectedItem = expectedItems[i];
+      const expectedCategory = expectedCategories[i];
+      const itemLabel = this.selectedItemsList.nth(i);
+      await expect(itemLabel).toHaveText(expectedItem);
+      const categoryLabel = this.selectedCategoryList.nth(i);
+      await expect(categoryLabel).toHaveText(expectedCategory);
+    }
+  }
+
+  // Sample details methods (shared across all modals)
+  async validateSelectedItemInSampleDetailsPage(itemName: string) {
+    // Wait for the sample details page to load
+    await this.dateTimeCollectedInputs.first().waitFor({ state: 'visible' });
+    
+    // Look for the item name in the sample details typography element
+    const itemInSampleDetails = this.sampleDetailsPanels.filter({ hasText: itemName });
+    await expect(itemInSampleDetails).toBeVisible();
+  }
+
+  // Set date/time collected for a specific test/panel (index allows targeting multiple inputs when multiple tests are selected)
+  async setDateTimeCollected(dateTime: string, index: number = 0) {
+    const input = this.dateTimeCollectedInputs.locator('input').nth(index);
+    await input.click();
+    await input.waitFor({ state: 'visible' });
+    await input.fill(dateTime);
+  }
+  
+  // Select first collected by option for a specific test/panel (index allows targeting multiple inputs when multiple tests are selected)
+  async selectFirstCollectedBy(index: number = 0) {
+    const input = this.collectedByInputs.nth(index);
+    await input.click();
+    const firstOptionLocator = this.collectedBySuggestionsList.locator('ul').locator('li').first();
+    await firstOptionLocator.click();
+    return await firstOptionLocator.textContent();
+  }
```

Comment:
```suggestion
  /**
   * Selects the first collected by option for a specific test/panel
   * @param index - The index of the input to select (allows targeting multiple inputs when multiple tests are selected)
   * @returns The text content of the first collected by option
   */
  async selectFirstCollectedBy(index: number = 0) {
    const input = this.collectedByInputs.nth(index);
    await input.click();
    const firstOptionLocator = this.collectedBySuggestionsList.locator('ul').locator('li').first();
    await firstOptionLocator.click();
    return await firstOptionLocator.textContent();
  }
```

Totally optional, but if you like JSdoc format like this is a nice convention for describing what the parameters of a function are used for. There's a shortcut to add one by just typing `/**` on top of a function